network/src/peer_manager/mod.rs

```diff
@@ -170,12 +170,10 @@ impl PeerManagerRequestSender {
             res_tx,
             timeout,
         };
-        self.inner
-            .push(
-                (peer_id, protocol),
-                PeerManagerRequest::SendRpc(peer_id, request),
-            )
-            .unwrap();
```

Comment:
I will argue that the `unwrap` is desirable. A `push` on this channel failing means that the PeerManager is, for some reason, no longer acceping outbound messages. This points to a bug/failure of the network module, something we can't possibly recover from. So it's best to crash right away instead of observing it at a distance.

Reply:
This is about clean destruction, unwrap here crashed the thread if another one is dropped first

Reply:
Ahh...I did realize that it could be a problem when dropping values in tests. Anyway we can find a middle ground?

Reply:
we can do sth like `if let Err(e) = .. { error!();}` inside this function